src/components/fields/Decimal/index.ts

```diff
@@ -63,43 +63,49 @@ class Decimal extends FormComponentBase {
     //  of any component that's a child of BridgeBase with a call to this.prepareForRender();
     this.prepareForRender();
 
+    // return if not visible
+    if (!this.bVisible) {
+      return nothing;
+    }
+
     // Handle and return if read only rendering
     if (this.bReadonly) {
-      return html`
-        <text-form
-          .pConn=${this.thePConn}
-          ?disabled=${this.bDisabled}
+      const theContent = html`
+        <lion-input-amount
+          id=${this.theComponentId}
+          ?readonly=${this.bReadonly}
           ?visible=${this.bVisible}
```

Comment:
Not sure it's necessary to set visible - we early return if it's not visible and the default for visible is true.